### ISSUE TITLE
port to a gdnative plugin

### ISSUE THREAD

**Issue by blurymind** [sanja-sa/gddragonbones#8]
Now that this module works so well and gdnative and godot3 has reached a stable level, it would be amazing if it could be ported to gdnative and made available  on godot's assets website as an addon people can just download without the need to recompile godot.

I think that would greatly increase its adoption across the community.

This however raises a number of questions:
- How hard is it to port a module to gdnative?
- Is there any valuable example/documentation that would help
The only example of a gdnative ported module on the asset website seems to be:
https://godotengine.org/asset-library/asset/150
https://github.com/BastiaanOlij/godot-openvr-asset

Perhaps we can get @BastiaanOlij to help?

- Are there any current limitations or problems with gdnative, apart of the poor documentation

**Comment by Ranoller** [comment]
I prefer it in module, performance is good, if this will be a core resource of a game (to handle a lot of the animations) it´s ok to have it in module. Any godot user that want to end their game have to learn to compile godot, there are configuration´s that are not posible without compiling. Using the store to visibility it´s ok, but the way that sanja-sa has chosen for this module is better than an external library, i think.

**Comment by blurymind** [comment]
@Ranoller I love the module too. It doesn't make compile time much longer.
I have tried in the past to convince godot developers to include a dragonbones, spine2d or a creature2d runtime with godot - giving examples of other engines that do (cocos creator, defold, appgamekit, etc etc) - but they have refused because they dont want to maintain them and instead create their own implementation..

With this module maturing feature wise and having such stability and quality - do you think @akien-mga and @reduz would reconsider including it in godot's master branch?

**Comment by Ranoller** [comment]
I can say to you exact words that reduz will say:
- That will never happend.

**Comment by blurymind** [comment]
@Ranoller yeah unfortunately he is very protective of Godot including third party libraries and now with gdnative he will always have the "use gdnative instead" excuse

Right now gdnative is the only way to get this inside godot editor- out of the box
